From 2f930866a84da48a26feb8f1fe4cfa18d9c48cec Mon Sep 17 00:00:00 2001
From: Lu Jialin <lujialin4@huawei.com>
Date: Mon, 17 Oct 2022 10:45:06 +0800
Subject: [PATCH 1072/1076] [Huawei] cgroup: Fix build error when
 CONFIG_CGROUP_WRITEBACK is n
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187773

-------------------------------

A build with CONFIG_CGROUP_WRITEBACK=n will produce the following errors

mm/memcontrol.c: In function ‘mem_cgroup_css_from_page’:
mm/memcontrol.c:426:6: error: implicit declaration of function ‘cgroup1_writeback_enabled’; did you mean ‘cgroup_writeback_unt’? [-Werror=implicit-function-declaration]
      cgroup1_writeback_enabled()))
      ^~~~~~~~~~~~~~~~~~~~~~~~~
      cgroup_writeback_umount

Fix it through define cgroup1_writeback_enabled() return false when
CONFIG_CGROUP_WRITEBACK=n

Fixes: 13b8199f0a47 ("cgroup: support cgroup writeback on cgroupv1")
Signed-off-by: Lu Jialin <lujialin4@huawei.com>
---
 include/linux/backing-dev.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/backing-dev.h b/include/linux/backing-dev.h
index 61aed9511150..ddbb0e6ff40f 100644
--- a/include/linux/backing-dev.h
+++ b/include/linux/backing-dev.h
@@ -396,6 +396,11 @@ static inline int inode_congested(struct inode *inode, int cong_bits)
 	return wb_congested(&inode_to_bdi(inode)->wb, cong_bits);
 }
 
+static inline bool cgroup1_writeback_enabled(void)
+{
+	return false;
+}
+
 #endif	/* CONFIG_CGROUP_WRITEBACK */
 
 static inline int inode_read_congested(struct inode *inode)
-- 
2.17.1

